plugins/modules/sda_fabric_devices_workflow_manager.py

```diff
@@ -3932,9 +4013,53 @@ def update_fabric_devices(self, fabric_devices):
                 self.log("Desired fabric device '{ip}' details (want): {requested_state}"
                          .format(ip=device_ip, requested_state=want_device_details), "DEBUG")
                 try:
-                    payload = {"payload": [want_device_details]}
-                    task_name = "add_fabric_devices"
-                    task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+                    device_roles = want_device_details.get("deviceRoles")
+                    if device_roles == ["CONTROL_PLANE_NODE"]:
```

Comment:
Can we make it simple? 

    route_distribution_protocol = item.get("route_distribution_protocol", "LISP_BGP")
    self.log(
        "Route distribution protocol set to: {route_distribution_protocol}".format(
            route_distribution_protocol=route_distribution_protocol
        ), "DEBUG"
    )

    valid_protocols = ["LISP_BGP", "LISP_PUB_SUB"]
    if route_distribution_protocol not in valid_protocols:
        self.msg = (
            "The 'route_distribution_protocol' must be one of {valid_protocols}."
            .format(valid_protocols=valid_protocols)
        )
        self.status = "failed"
        self.log(
            "Invalid route distribution protocol: {route_distribution_protocol}. Allowed values: {valid_protocols}"
            .format(route_distribution_protocol=route_distribution_protocol, valid_protocols=valid_protocols),
            "ERROR"
        )
        return self

    payload = {